src/emailservice/Dockerfile

```diff
@@ -1,27 +1,52 @@
+FROM python:2.7-alpine3.8 as base
 
+FROM base as builder
 
-# Use the grpc.io provided Python image as the base image
-FROM grpc/python:1.0
+# gRPC deps
+RUN apk add --update --no-cache \
+    gcc \
+    linux-headers \
+    make \
+    musl-dev \
+    python-dev \
+    g++ \
+    cairo-dev \
+    cairo \
+    openssl-dev \ 
+    gobject-introspection-dev
 
-# download the grpc health probe
+ENV GRPC_PYTHON_VERSION 1.15.0
+
+# Install grpcio
+RUN python -m pip install --upgrade pip
+RUN pip install grpcio==${GRPC_PYTHON_VERSION} grpcio-tools==${GRPC_PYTHON_VERSION}
+
+# get packages
+COPY requirements.txt .
+
+RUN pip install -r requirements.txt
+
+
+FROM base as final
+
+# Download the grpc health probe
 RUN GRPC_HEALTH_PROBE_VERSION=v0.1.0-alpha.1 && \
     wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
     chmod +x /bin/grpc_health_probe
 
-# show python logs as they occur
+# Show python logs as they occur
 ENV PYTHONUNBUFFERED=0
 
-# install pip for python3
-RUN apt-get -qqy update && \
-        apt-get -qqy install python3-pip
-
-# get packages
 WORKDIR /email_server
-COPY requirements.txt .
-RUN pip3 install -r requirements.txt
+
+# Grab packages from builder
+COPY --from=builder /usr/local/lib/python2.7/ /usr/local/lib/python2.7/
+
+# Need libstdc++ for grpc
+RUN apk add --no-cache libstdc++
 
 # Add the application
 COPY . .
 
 EXPOSE 8080
-ENTRYPOINT [ "python3", "email_server.py" ]
+ENTRYPOINT [ "python", "email_server.py" ]
```

Comment:
Can't we keep it python3? Did you refactor the app?

Reply:
I downgraded to python 2.7 as the gRPC portion was using python 2.7, and in my testing the app worked on both. Will attempt to upgrade the gRPC section to python 3

Reply:
Upgraded to python 3